src/scripting/scriptingService.ts

```diff
@@ -0,0 +1,95 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import SqlToolsServiceClient from '../languageservice/serviceclient';
+import ConnectionManager from '../controllers/connectionManager';
+import { ScriptingRequest, ScriptingParams, ScriptOperation, ScriptingObject, ScriptOptions } from '../models/contracts/scripting/scriptingRequest';
+import { TreeNodeInfo } from '../objectExplorer/treeNodeInfo';
+import VscodeWrapper from '../controllers/vscodeWrapper';
+
+export class ScriptingService {
+
+    private _client: SqlToolsServiceClient;
+
+    constructor(
+        private _connectionManager: ConnectionManager,
+        private _vscodeWrapper: VscodeWrapper
+    ) {
+        this._client = this._connectionManager.client;
+    }
+
+    // map for the version of SQL Server (default is 140)
+    readonly scriptCompatibilityOptionMap = {
```

Comment:
Bleh...we should really be supporting 150. But in general I'm not a fan of us having these hard coded values everywhere since it makes vBump a huge pain. Nothing I expect you to fix right now but if you could be thinking about ways to make this not be duplicated as much that'd be good.